app/activeproject.cpp

```diff
@@ -364,3 +364,18 @@ void ActiveProject::setActiveLayer( QgsMapLayer *layer ) const
     mAppSettings.setDefaultLayer( mActiveLayer.layerName() );
   }
 }
+
+bool ActiveProject::reloadPreservingExtent()
```

Comment:
is there any situation when you do not want to preserve extent? if yes, maybe add `preserveExtent=True` to `reload()` function?

Reply:
From my point of view we should preserve extent all the time, but I can be not aware about all possible use cases.

Reply:
A function called `reloadProject` is currently not being used anywhere (as we removed the duplicated signal in MerginApi). I'd say let's have only the `reloadProject` function that is going to preserve the extent (as it is only used on one place)